components/popover/index.js

```diff
@@ -31,54 +32,42 @@ const { ESCAPE } = keycodes;
  * @type {String}
  */
 const SLOT_NAME = 'Popover';
-const isMobile = () => window.innerWidth < 782;
 
 class Popover extends Component {
 	constructor() {
 		super( ...arguments );
 
 		this.focus = this.focus.bind( this );
-		this.bindNode = this.bindNode.bind( this );
 		this.getAnchorRect = this.getAnchorRect.bind( this );
-		this.setOffset = this.setOffset.bind( this );
-		this.throttledSetOffset = this.throttledSetOffset.bind( this );
+		this.computePopoverPosition = this.computePopoverPosition.bind( this );
+		this.throttledComputePopoverPosition = this.throttledComputePopoverPosition.bind( this );
 		this.maybeClose = this.maybeClose.bind( this );
 
-		this.nodes = {};
+		this.contentNode = createRef();
+		this.anchorNode = createRef();
 
 		this.state = {
-			forcedYAxis: null,
-			forcedXAxis: null,
+			popoverLeft: null,
+			popoverTop: null,
+			yAxis: 'top',
+			xAxis: 'center',
+			contentHeight: null,
+			contentWidth: null,
 			isMobile: false,
+			popoverSize: null,
 		};
 	}
 
 	componentDidMount() {
-		this.setOffset();
-		this.setForcedPositions();
+		const popoverSize = this.updatePopoverSize();
+		this.computePopoverPosition( popoverSize );
 		this.toggleWindowEvents( true );
-		this.focus();
+		setTimeout( () => this.focus() );
```

Comment:
I'd love to get rid of this timeout but for some mysterious reason, it's not working properly without it.

 - Try removing the timeout
 - Open the inserter
 - The search input is not focused.

I tried debugging and for some reason the call to `input.focus()` is not focusing the input (console logging the `document.actiiveElement`.

Reply:
I tried to debug, and I did not find a good solution to this, It looks like although the parent component was mounted the child is not right away in a state that we can focus, this discussion seems related https://stackoverflow.com/questions/35522220/react-ref-with-focus-doesnt-work-without-settimeout-my-example.

What complicates, even more, the things is that if we use withSafeTimeout, we create the z-index bug ( i tested it again).

Maybe we can move the setTimeout inside the focus function so inside the timeout we don't reference this, props, etc.. we reference the dom node something like:
```
		const focusNode = ( domNode ) => setTimeout( () => domNode.focus() );
		const domNode = this.contentNode.current;
		// Find first tabbable node within content and shift focus, falling
		// back to the popover panel itself.
		const firstTabbable = focus.tabbable.find( domNode )[ 0 ];
		focusNode( firstTabbable ? firstTabbable : domNode );
```

Reply:
I don't expect this to create any issue regarding "this" because it's during mouting but even though, I like your proposal let's move it to `focus` and add a comment.

Reply:
Maybe we can also add a comment specifying why we are using a set timeout, and why withSafeTimeout was not possible to use.